Polyhedron/demo/Polyhedron/Scene_polyhedron_item.cpp

```diff
@@ -631,19 +642,34 @@ Scene_polyhedron_item::Scene_polyhedron_item(const Polyhedron& p)
       erase_next_picked_facet_m(false),
       plugin_has_set_color_vector_m(false)
 {
-    //setItemIsMulticolor(true);
     cur_shading=FlatPlusEdges;
     is_selected=true;
+    textItems = new TextListItem(this);
     init();
     nb_facets = 0;
     nb_lines = 0;
     invalidate_buffers();
+	targeted_id = NULL;
 }
 
 Scene_polyhedron_item::~Scene_polyhedron_item()
 {
     delete_aabb_tree(this);
     delete poly;
+    QGLViewer* viewer = *QGLViewer::QGLViewerPool().begin();
+    if(viewer)
+    {
+      CGAL::Three::Viewer_interface* v = qobject_cast<CGAL::Three::Viewer_interface*>(viewer);
+
+      //Clears the targeted Id
+      targeted_id = NULL;
+      delete targeted_id;
+      //Remove textitems
+      v->textRenderer->removeTextList(textItems);
+      textItems =0;
+      delete textItems;
+    }
```

Comment:
There is a memory leak, here: line 670 calls `delete 0`, which does nothing. It should be:
```C++
delete textItems;
textItems = 0;
```
and `textItems = 0;` is useless in a destructor: as the object is being destroyed, the value of `textItems` does no long matter.